devito/ir/iet/nodes.py

```diff
@@ -771,6 +780,34 @@ def exprs(self):
         return self.body
 
 
+class Prodder(Call):
+
+    """
+    A Call promoting asynchronous progress, to minimize latency.
+
+    Example use cases:
+
+        * To trigger asynchronous progress in the case of distributed-memory
+          parallelism.
+        * Software prefetching.
+    """
```

Comment:
Is it possible to expand this docstring a bit? e.g. in what way are these asynchronous calls really different from standard calls - in terms of generated code?

Reply:
indeed they are not -- but Prodders may need to be distinguished from Calls as they may be subject of optimisation (e.g., code motion).

Also -- and this is implicit in the description -- a Prodder, when invoked, is guaranteed not to alter state (which ensures optimizations such as code motion are safe)